### PR TITLE
[WIP] Add check-changelog.yml workflow

### PR DESCRIPTION
**Pull request by zeme-iohk**

### CODE REVIEW

File: .github/workflows/check-changelog.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        with:
+          files: '**/changelog.d/**'
+
+      - name: Enforce New File or DISABLE_CHANGELOG_CHECK
+        uses: actions/github-script@v6
+        with:
+          script: | 
+            function shouldCheckChangelog() {
+              const prDescription = '${{ github.event.pull_request.body }}';
+              return !prDescription.includes("DISABLE_CHANGELOG_CHECK");
```
Comment (michaelpj) [inline .github/workflows/check-changelog.yml:23]: Maybe we should use a label for this? Also makes it easier to see elsewhere.

Comment (zeme-iohk) [reply .github/workflows/check-changelog.yml:23]: I'm not sure what labels are in this context

Comment (zeme-iohk) [reply .github/workflows/check-changelog.yml:23]: Or so we'd look for a label named "Requires Changelog"? or "Does not Require Changelog"?

Comment (michaelpj) [reply .github/workflows/check-changelog.yml:23]: I think we want to label the exceptional case, so we'd have a "No Changelog Required" label and the workflow would check for that when deciding whether to complain. We would also need to rerun the workflow if the labels change.

File: .gitignore [hunk 2]
```diff
@@ -89,6 +89,7 @@ plutus-playground-server/playground.yaml
 plutus-pab/plutus-pab.yaml
 
 # Misc
+act-event.json
```
Comment (zeme-iohk) [inline .gitignore:92]: We need this for testing GH actions locally

**Comment by zeme-iohk** [comment]
I've tested manually by adding & removing the label, and by committing and deleting a file in plutus-core/changelog.d

File: .github/workflows/pre-commit-check.yml [hunk 3]
```diff
@@ -1,6 +1,9 @@
 name: Pre Commit Check
 on:
   pull_request:
+    types:
+      - opened
+      - synchronize
```
Comment (michaelpj) [inline .github/workflows/pre-commit-check.yml:6]: Do we need this?

**Review by michaelpj — APPROVED** [review]
LGTM.

**Comment by michaelpj** [comment]
Oh, probably needs to be documented in the changelog section of CONTRIBUTING

File: .github/workflows/pre-commit-check.yml (hunk 3 above)
Comment (zeme-iohk) [reply .github/workflows/pre-commit-check.yml:6]: I should have added a comment: with this, the "Pre Commit Check" will *not* be run when we label/unlabel the PR, which I trust is what we want

**Comment by zeme-iohk** [comment]
> Oh, probably needs to be documented in the changelog section of CONTRIBUTING

Done

**Pull request merged**